File: Ejercicio8.py
# Precios de Arcade - Python

def precio_boleto(edad):
    """Devuelve el precio del boleto según la edad.
    Lanza ValueError si la edad no es un entero no negativo."""
    if not isinstance(edad, int):
        raise ValueError("La edad debe ser un entero.")
    if edad < 0:
        raise ValueError("La edad no puede ser negativa.")
    if edad <= 3:
        return 0.0
    if edad <= 12:
        return 5.0
    if edad <= 17:
        return 7.0
    if edad <= 64:
        return 12.0
    return 8.0  # 65+

def main():
    try:
        entrada = input("Ingresa la edad del cliente: ").strip()
        edad = int(entrada)
    except ValueError:
        print("Entrada inválida: por favor ingresa un número entero no negativo.")
        return

    try:
        precio = precio_boleto(edad)
    except ValueError as e:
        print("Error:", e)
        return

    if precio == 0.0:
        print("Precio: Entrada gratuita.")
    else:
        print(f"Precio del boleto: ${precio:.2f}")

if __name__ == "__main__":
    main()

File: Ejercicio8.hs
-- Precios de Arcade - Haskell
import Text.Read (readMaybe)

-- Devuelve Either String Double: Left = error, Right = precio
precioBoleto :: Int -> Either String Double
precioBoleto edad
  | edad < 0  = Left "Edad inválida: no puede ser negativa."
  | edad <= 3 = Right 0.0
  | edad <= 12 = Right 5.0
  | edad <= 17 = Right 7.0
  | edad <= 64 = Right 12.0
  | otherwise  = Right 8.0  -- 65+

mostrarPrecio :: Int -> String
mostrarPrecio edad = case precioBoleto edad of
  Left err   -> err
  Right 0.0  -> "Precio: Entrada gratuita."
  Right p    -> "Precio del boleto: $" ++ show p

main :: IO ()
main = do
  putStrLn "Ingresa la edad del cliente:"
  line <- getLine
  case readMaybe line :: Maybe Int of
    Just edad -> putStrLn (mostrarPrecio edad)
    Nothing   -> putStrLn "Entrada inválida: por favor ingresa un número entero no negativo."
